### PR TITLE
feat(SideNav): init

### PR DESCRIPTION
**Pull request by cdaringe**
# problem statement

- `<LeftNav />`:
   - doesn't scale well
   - has excess classes, inconsistent classNams
   - is strongly opinionated to the _left_ (which this PR doesn't add _right_ support, but allows it to be extended to easily)

# solution

- replace with `<SideNav />`, maintaining the status quo functionality of `LeftNav`
- add a nice hover transition

### CODE REVIEW

File: src/components/SideNav/side-nav.css [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    font-size: 2rem;
+    justify-content: center;
+    padding: 10px 0 10px 0;
+    transition: opacity 400ms;
+  }
+
+  & .nav__item img {
+    height: 50px;
+    width: 50px;
+  }
```
Comment (cdaringe) [inline src/components/SideNav/side-nav.css]: :eyes: tomorrow, note to self, drop. work w/ scaling a tad more

File: src/components/SideNav/side-nav.css [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  & .icon_container--active {
+    opacity: 1;
+    & :hover{
```
Comment (cdaringe) [inline src/components/SideNav/side-nav.css]: drop

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereviewtstficxkrkpnwhr.surge.sh](http://surgereviewtstficxkrkpnwhr.surge.sh)

File: src/components/SideNav/SideNavMenuItem.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  const { active, image, className, children, ...rest } = props
+  const iconClasses = cx('icon_container', `icon_container--${active ? 'active' : 'inactive'}`)
+  return (
+    <li className={cx('side-nav__item', className)} {...rest}>
+      {active ? <div className='side-nav_border--active' /> : null}
+      <div className={iconClasses}>
+        {image ? <img src={image} /> : null}
+        {children}
+      </div>
+    </li>
```
Comment (cdaringe) [inline src/components/SideNav/SideNavMenuItem.jsx]: i ended up _not_ trimming down the amount of DOM, but i did simplify & hack out CSS

**Comment by wa11-e** [comment]
Woohoo! new surge deployment available for viewing! :tada: [surgereviewyfzwpfnyyjtyhhs.surge.sh](http://surgereviewyfzwpfnyyjtyhhs.surge.sh)

**Comment by andrewleyva** [comment]
Yeah.. why was LeftNav ever a thing. Good call. This makes this component far more useful.

**Comment by cdaringe** [comment]
yep.  1st commit has BREAKING CHANGE text in it

**Pull request merged**